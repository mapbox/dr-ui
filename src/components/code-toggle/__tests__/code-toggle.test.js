import React from 'react';
import renderer from 'react-test-renderer';
import { testCases } from './code-toggle-test-cases.js';

describe('code-toggle', () => {
  describe(testCases.basic.description, () => {
    let testCase;
    let wrapper;
    let tree;

    beforeEach(() => {
      testCase = testCases.basic;
      wrapper = renderer.create(testCase.element);
      tree = wrapper.toJSON();
    });

    test('renders as expected', () => {
      expect(tree).toMatchSnapshot();
    });
  });

  describe(testCases.objectiveC.description, () => {
    let testCase;
    let wrapper;
    let tree;

    beforeEach(() => {
      testCase = testCases.objectiveC;
      wrapper = renderer.create(
        React.createElement(testCase.component, testCase.props)
      );
      tree = wrapper.toJSON();
    });

    test('renders as expected', () => {
      expect(tree).toMatchSnapshot();
    });
  });

  describe(testCases.objectiveC2.description, () => {
    let testCase;
    let wrapper;
    let tree;

    beforeEach(() => {
      testCase = testCases.objectiveC2;
      wrapper = renderer.create(
        React.createElement(testCase.component, testCase.props)
      );
      tree = wrapper.toJSON();
    });

    test('renders as expected', () => {
      expect(tree).toMatchSnapshot();
    });
  });
});
