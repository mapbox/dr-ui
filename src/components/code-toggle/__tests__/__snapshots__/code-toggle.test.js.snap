// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`code-toggle Java preferred renders as expected 1`] = `
<div
  className="relative "
>
  <fieldset>
    <div
      aria-required={true}
      className="toggle-group round-full bg-blue py3 px3"
      data-test="three-input"
      id="three"
      role="radiogroup"
    >
      <label
        className="toggle-container "
      >
        <input
          autoFocus={false}
          checked={false}
          name="three"
          onChange={[Function]}
          type="radio"
          value="javascript"
        />
        <div
          className="txt-s py3 toggle--white toggle--active-blue toggle"
        >
          JavaScript
        </div>
      </label>
      <label
        className="toggle-container "
      >
        <input
          autoFocus={false}
          checked={true}
          name="three"
          onChange={[Function]}
          type="radio"
          value="java"
        />
        <div
          className="txt-s py3 toggle--white toggle--active-blue toggle"
        >
          Java
        </div>
      </label>
      <label
        className="toggle-container "
      >
        <input
          autoFocus={false}
          checked={false}
          name="three"
          onChange={[Function]}
          type="radio"
          value="swift"
        />
        <div
          className="txt-s py3 toggle--white toggle--active-blue toggle"
        >
          Swift
        </div>
      </label>
      <label
        className="toggle-container "
      >
        <input
          autoFocus={false}
          checked={false}
          name="three"
          onChange={[Function]}
          type="radio"
          value="objectiveC"
        />
        <div
          className="txt-s py3 toggle--white toggle--active-blue toggle"
        >
          Objective-C
        </div>
      </label>
    </div>
  </fieldset>
  <div
    role="alert"
  />
</div>
`;

exports[`code-toggle Objective C preferred renders as expected 1`] = `
<div
  className="relative "
>
  <fieldset>
    <div
      aria-required={true}
      className="toggle-group round-full bg-blue py3 px3"
      data-test="two-input"
      id="two"
      role="radiogroup"
    >
      <label
        className="toggle-container "
      >
        <input
          autoFocus={false}
          checked={false}
          name="two"
          onChange={[Function]}
          type="radio"
          value="swift"
        />
        <div
          className="txt-s py3 toggle--white toggle--active-blue toggle"
        >
          Swift
        </div>
      </label>
      <label
        className="toggle-container "
      >
        <input
          autoFocus={false}
          checked={true}
          name="two"
          onChange={[Function]}
          type="radio"
          value="objectiveC"
        />
        <div
          className="txt-s py3 toggle--white toggle--active-blue toggle"
        >
          Objective-C
        </div>
      </label>
    </div>
  </fieldset>
  <div
    role="alert"
  />
</div>
`;

exports[`code-toggle Swift preferred renders as expected 1`] = `
<div
  className="relative "
>
  <fieldset>
    <div
      aria-required={true}
      className="toggle-group round-full bg-blue py3 px3"
      data-test="one-input"
      id="one"
      role="radiogroup"
    >
      <label
        className="toggle-container "
      >
        <input
          autoFocus={false}
          checked={true}
          name="one"
          onChange={[Function]}
          type="radio"
          value="swift"
        />
        <div
          className="txt-s py3 toggle--white toggle--active-blue toggle"
        >
          Swift
        </div>
      </label>
      <label
        className="toggle-container "
      >
        <input
          autoFocus={false}
          checked={false}
          name="one"
          onChange={[Function]}
          type="radio"
          value="objectiveC"
        />
        <div
          className="txt-s py3 toggle--white toggle--active-blue toggle"
        >
          Objective-C
        </div>
      </label>
    </div>
  </fieldset>
  <div
    role="alert"
  />
</div>
`;
